### ISSUE TITLE
Could not load file or assembly 'TypeShape, Version=8.0.1.0'

### ISSUE THREAD

**Issue by queil** [demystifyfp/FsConfig#17]
### Description

Retrieving the config fails with `Could not load file or assembly 'TypeShape, Version=8.0.1.0,...`

### Repro steps

1. Create a new project referencing FsConfig
2. Create a config record type and use the default match expression from the docs to retrieve the config

### Expected behavior

Config gets successfully retrieved.

### Actual behavior

The following exception gets thrown:
```
Unhandled exception. System.IO.FileLoadException: Could not load file or assembly 'TypeShape, Version=8.0.1.0, Culture=neutral, PublicKeyToken=null'. The located assembly's manifest definition does not match the assembly reference. (0x80131040)
File name: 'TypeShape, Version=8.0.1.0, Culture=neutral, PublicKeyToken=null'
   at FsConfig.Core.parseInternal[T](IConfigReader configReader, FSharpFunc`2 fieldNameCanonicalizer, FieldValueParseArgs args)
   at Program.main(String[] argv) in
```

### Known workarounds

Add TypeShape directly to the project:
```
    <PackageReference Include="TypeShape" Version="8.0.1" />
```

### Related information 

* Operating system: Ubuntu 18.04
* FsConfig: 2.1.5
* .NET Core 3.1.101

**Comment by demystifyfp** [comment]
Once again sorry for the delayed response @queil. I am not able to reproduce the issue. 

The TypeShape dependency is explicitly mentioned and it is correctly showing up in the [NuGet package ](https://www.nuget.org/packages/FsConfig/) as well.

Can you share me a sample project using which I can reproduce the issue?

**Comment by queil** [comment]
@demystifyfp A repro Dockerfile:

```Dockerfile
FROM mcr.microsoft.com/dotnet/core/sdk:3.1.201-bionic
WORKDIR /app
RUN dotnet new console -lang f# && dotnet add package FsConfig && echo '\n\
open System \n\
open FsConfig\n\
[<EntryPoint>]\n\
let main argv =\n\
  EnvConfig.Get<string>() |> ignore\n\
  0' \ 
> Program.fs && dotnet run
```
Obviously `string` does not make much sense there but this is only to get a minimal repro. It fails the same way with a record.

**Comment by demystifyfp** [comment]
It looks like I need some more time to troubleshoot this. I will get back to you this weekend.

**Comment by demystifyfp** [comment]
Hi @queil,

Sorry for the delayed response.

It's working well with `paket`. 

I guess with it is something to do with `dotnet` core package resolution. I don't much expertise on it. So, I am not sure how can I help here!

**Comment by digital-stoic** [comment]
Same issue here. Quick workaround: `dotnet add package typeshape`

**Comment by jukkhop** [comment]
Seeing this issue too, the suggested workarounds do not seem to work. The exact error I have is:

```
Resolved assembly's simple name should be the same as of the requested assembly.

System.Private.CoreLib
  at System.Runtime.Loader.AssemblyLoadContext.ValidateAssemblyNameWithSimpleName(Assembly assembly, String requestedSimpleName)
  at System.Runtime.Loader.AssemblyLoadContext.ResolveUsingLoad(AssemblyName assemblyName)
  at System.Runtime.Loader.AssemblyLoadContext.Resolve(IntPtr gchManagedAssemblyLoadContext, AssemblyName assemblyName)
```

**Comment by Rajivhost** [comment]
Hi all,

I have the same issue as below.

> Seeing this issue too, the suggested workarounds do not seem to work. The exact error I have is:
> 
> ```
> Resolved assembly's simple name should be the same as of the requested assembly.
> 
> System.Private.CoreLib
>   at System.Runtime.Loader.AssemblyLoadContext.ValidateAssemblyNameWithSimpleName(Assembly assembly, String requestedSimpleName)
>   at System.Runtime.Loader.AssemblyLoadContext.ResolveUsingLoad(AssemblyName assemblyName)
>   at System.Runtime.Loader.AssemblyLoadContext.Resolve(IntPtr gchManagedAssemblyLoadContext, AssemblyName assemblyName)
> ```

**Comment by tamizhvendan** [comment]
@queil  @jukkhop @matieux @Rajivhost 

I have recently changed my Mac and I am unable to do a fresh release due to [this open issue](https://github.com/fsprojects/ProjectScaffold/issues/349) in ProjectScaffold (FsConfig based on this) from my new Mac. 

As suggested in [this comment](https://github.com/fsprojects/ProjectScaffold/issues/349#issuecomment-626373929), I have to either migrate to MiniScaffold or Waypoint. 

Currently, due to official & personal commitments, I am not in a position to do this migration. So, if you would like to contribute to this, I am open to accept your pull request.

**Comment by outergod** [comment]
Same problem here.

```
Unhandled exception. System.IO.FileLoadException: Could not load file or assembly 'TypeShape, Version=8.0.1.0, Culture=neutral, PublicKeyToken=null'. The located assembly's manifest definition does not match the assembly reference. (0x80131040)
File name: 'TypeShape, Version=8.0.1.0, Culture=neutral, PublicKeyToken=null'
   at FsConfig.Core.parseInternal[T](IConfigReader configReader, FSharpFunc`2 fieldNameCanonicalizer, FieldValueParseArgs args)
[...]
```

The suggested workaround however works.

**Comment by queil** [comment]
@tamizhvendan I've just published version 3.0.0. The problem looks solved now. Tested with the following `Dockerfile`:
```
FROM mcr.microsoft.com/dotnet/sdk:5.0.102-ca-patch-buster-slim
WORKDIR /app
RUN dotnet new console -lang f# && dotnet add package FsConfig && echo '\n\
open System \n\
open FsConfig\n\
[<EntryPoint>]\n\
let main argv =\n\
  EnvConfig.Get<string>() |> ignore\n\
  printfn "- OK -"\n\
  0' \
> Program.fs && dotnet run
```

**Comment by tamizhvendan** [comment]
Thanks a ton, @queil, for all the hard work and get it working. Sorry for the delayed response. The last few weeks were just crazy both at work and in life.
